samples/nrf9160/nrf_cloud_mqtt_multi_service/src/fota_support.c

```diff
@@ -13,10 +13,13 @@
 
 LOG_MODULE_REGISTER(fota_support, CONFIG_MQTT_MULTI_SERVICE_LOG_LEVEL);
 
-#if defined(CONFIG_NRF_CLOUD_FOTA_FULL_MODEM_UPDATE)
+#if (IS_ENABLED(CONFIG_NRF_CLOUD_FOTA_FULL_MODEM_UPDATE) && \
+	!IS_ENABLED(CONFIG_DFU_TARGET_FULL_MODEM_USE_EXT_PARTITION))
 /* Full modem FOTA requires external flash to hold the full modem image.
  * Below is the external flash device present on the nRF9160 DK version
  * 0.14.0 and higher.
+ * If an external partition is used, it is not necessary to specify the
+ * flash device.
  */
 #define FULL_MODEM_FOTA_FLASH_DEVICE DEVICE_DT_GET_ONE(jedec_spi_nor)
```

Comment:
This is out-of-scope for the PR but I think maybe migrating this to check for `DT_CHOSEN(nordic_pm_ext_flash)` would probably be a good idea.

Reply:
created CIA-882 for this